core/NameRef.h

```diff
@@ -82,14 +96,29 @@ class NameRef final : private DebugOnlyCheck<NameRefDebugCheck> {
     // generated by tools/generate_names.cc
     struct WellKnown {};
 
-    constexpr NameRef(WellKnown, u4 id) : _id{id} {}
+    constexpr NameRef(WellKnown, NameKind kind, u4 id) : _id{id | (static_cast<u4>(kind) << ID_BITS)} {
+        // If this fails, the symbol table is too big :(
+        ENFORCE_NO_TIMER(id <= ID_MASK);
```

Comment:
Is the decision to not mask off id intentional? That case would be caught by the enforce, but only if it shows up in a test somewhere. (I'm not really sure what a great way to test this would be, perhaps a one off test in c++ for name refs?)

Reply:
Not intentional. I can mask it off.

Reply:
Done!